### ISSUE TITLE
Autocorrect bug when completing files names in Code OSS-based IDEs

### ISSUE THREAD

**Issue by jthomasmock** [pappasam/jedi-language-server#325]
When using Jedi as the Language Server in Code OSS-based IDEs or in VS Code, auto-complete of file names at the `.` is incorrect. 

The text is auto-completed at the `.` instead of replacing the prior text.

Thank you for your time in reviewing this issue.

As an example, see the video below in VS Code with Jedi as the Language Server:

https://github.com/user-attachments/assets/3b6cf2eb-f419-4bfa-82e1-e4356745b3c1

moved over from: https://github.com/davidhalter/jedi/issues/2032

**Comment by davidhalter** [comment]
@pappasam Do you know of similar behavior? I haven't seen this or similar issues in the Jedi issue tracker (and there must be an insane amount of people using it).

**Comment by dimbleby** [comment]
I have never seen this in neovim, possibly this is an editor bug all along.  Someone would have to get trace of the lsp messages to decide.

Aside: I see quite a few bugs being punted around vscode and this repository and similar; often - as here - without anyone providing any evidence for where it actually belongs.  Perhaps editors could be encourage to make it easier to get useful diagnostics.

While I'm here: I actually wish jedi didn't provide filename completions at all, it seems quite out of scope for a python-analysis tool - and per #270 it's a bit too keen.

**Comment by pappasam** [comment]
@dimbleby yeah, it seems like a simple solution here might be for us to simply filter out filename completions coming from Jedi?

**Comment by jthomasmock** [comment]
Thanks for the discussion so far! I'm not seeing anything useful in the VS Code output for Python Language Server.

For what it's worth, Code OSS or VS Code w/ Pylance doesn't have built-in filename completions, but some extensions can add it so that could be a suggestion if y'all do allow disabling file completions?
- https://open-vsx.org/extension/ionutvmi/path-autocomplete
- https://open-vsx.org/extension/christian-kohler/path-intellisense

**Comment by davidhalter** [comment]
The issue here might be connected to https://github.com/pappasam/jedi-language-server/blob/1eeca008b17b13bb98d0d7e9c09a3fd5a3442082/jedi_language_server/jedi_utils.py#L409, because some characters are stripped, which might lead to a weird interaction with VSCode.

I was able to reproduce it in VSCode. It happens only after a dot in a string (i.e completing `"setup` is fine, but `"setup.` is not. Jedi looks fine:

```
>>> jedi.Script('"setup.').complete()
[<Completion: setup.cfg">, <Completion: setup.py">]
>>> jedi.Script('"setup.').complete()[0]
<Completion: setup.cfg">
>>> jedi.Script('"setup.').complete()[0].name
'setup.cfg"'
>>> jedi.Script('"setup.').complete()[0].complete
```

**Comment by pappasam** [comment]
Should be resolved in the latest release: https://github.com/pappasam/jedi-language-server/releases/tag/v0.42.0
